apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/AggregateQueryTest.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Firestore.IntegrationTests.Models;
+using Google.Cloud.Firestore.V1;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Firestore.AggregateField;
+
+namespace Google.Cloud.Firestore.IntegrationTests;
+
+[Collection(nameof(FirestoreFixture))]
+public class AggregateQueryTest
```

Comment:
Depending on the result, both `snapshot.GetValue<double>(...)` and `snapshot.GetValue<long>(...)` can be lossy. It might be worth adding a few tests to determine if the caller can get the precise result using `snapshot.GetValue<object>(...)` (or similar).

```
// `a` sums to a value that cannot be represented precisely as Int64
// `b` sums to an integer value that cannot be represented precisely as a double
addDoc({a: 0.1, b: (Int)2^53})
addDoc({a: 0.1, b: 1})
s = collection.aggregate(sum("a"), sum("b"))

var sumA = s.GetValue<Object>("sum_a")
Assert.Equal(sumA.GetType(), typeof(Double));

// Expected result from the server is an integer only if all input
// is integer values and the result does not overflow Int64
var sumB = s.GetValue<Object>("sum_b")
Assert.Equal(sumB.GetType(), typeof(Double));

Reply:
Right - yes, that makes sense to add as another integration test.